### PR TITLE
Add Azure Core OkHttp Javadocs 

### PR DESCRIPTION
**Pull request by g2vinay**
Adds class and package level javadoc for azure-core-http-okhttp package.

### CODE REVIEW

File: sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/OkHttpAsyncHttpClientBuilder.java [hunk 1]
```diff
@@ -26,6 +26,78 @@
 
 /**
  * Builder class responsible for creating instances of {@link com.azure.core.http.HttpClient} backed by OkHttp.
+ * Please be aware that client built from this builder can support synchronously and asynchronously call of sending
+ * request. Use {@link com.azure.core.http.HttpClient#sendSync(HttpRequest, Context)} to send the provided request
```
Comment (billwert) [inline sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/OkHttpAsyncHttpClientBuilder.java:30]:
```suggestion
 * Please be aware that client built from this builder can support sending requests synchronously or asynchronously.
 * request. Use {@link com.azure.core.http.HttpClient#sendSync(HttpRequest, Context)} to send the provided request
```

File: sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/OkHttpAsyncHttpClientBuilder.java [hunk 2]
```diff
@@ -26,6 +26,78 @@
...
+ *         .build&#40;&#41;;
+ * </pre>
+ * <!-- end com.azure.core.http.okhttp.OkHttpAsyncHttpClientBuilder#connectionTimeout -->
+ *
+ * <p><strong>Building a new HttpClient instance with HTTP/2 Support.</strong></p>
+ *
+ * <p>Configuring the OkHttp client with HTTP/2 support is relevant when you want to take advantage of the benefits
+ * provided by the HTTP/2 protocol. HTTP/2 is the next version of the HTTP network protocol, designed to improve the
+ * performance of web applications by allowing multiple requests and responses to be multiplexed over a
+ * single connection.</p>
```
Comment (billwert) [inline sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/OkHttpAsyncHttpClientBuilder.java:78]: this doesn't feel like the place for a discussion of the benefits of HTTP/2.

File: sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/package-info.java [hunk 3]
```diff
@@ -2,6 +2,126 @@
 // Licensed under the MIT License.
 
 /**
- * Package containing OkHttp HTTP client plugin for azure-core.
+ * <p><a href="https://learn.microsoft.com/en-us/java/api/overview/azure/core-http-okhttp-readme?view=azure-java-stable">
+ * Azure Core Http OkHttp</a> client library is a plugin for the azure-core HTTP client API. It allows you to use OkHttp
```
Comment (billwert) [inline sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/package-info.java:6]: link to the azure-core HTTP conceptual doc?

File: sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/OkHttpAsyncHttpClientBuilder.java (hunk 2 above)
Comment (g2vinay) [reply sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/OkHttpAsyncHttpClientBuilder.java:78]: I'll remove the brief HTTP/2 description and link out to its conceptual doc.

**Review by billwert — APPROVED** [review]

**Pull request merged**